src/mpl2/src/object.cpp

```diff
@@ -1031,9 +1023,92 @@ void HardMacro::updateDb(float pitch_x, float pitch_y, odb::dbBlock* block)
       = std::round(float(micronToDbu(origin_y, dbu_)) / manufacturing_grid_)
         * manufacturing_grid_;
 
-  inst_->setOrigin(snap_origin_x, snap_origin_y);
-  inst_->setOrient(orientation_);
-  inst_->setPlacementStatus(odb::dbPlacementStatus::LOCKED);
+  const odb::Point snap_origin(snap_origin_x, snap_origin_y);
+
+  return snap_origin;
+}
+
+// Compute for each pin: layer offset, pitch and pin width
+void HardMacro::computeCoordinateSpacingParameters(
```

Comment:
Why the `Coordinate` word in the function names? Perhaps `computeDirectionSpacingParameters` is more precise.

Reply:
The intent was to make clear that the the function works for one coordinate.
I agree with you. Direction is more precise.